packages/react-component-library/src/components/NumberInputE/NumberInputE.test.tsx

```diff
@@ -262,44 +272,74 @@ describe('NumberInputE', () => {
       })
     })
 
-    describe('and the user types a fractional value', () => {
+    describe('and the user types a negative value', () => {
       beforeEach(() => {
-        userEvent.type(input, '100.1')
+        userEvent.type(input, '-100')
       })
 
-      assertInputValue('100.1')
-      assertOnChangeCall(100.1, 5)
+      assertInputValue('-100')
+      assertOnChangeCall(-100, 4)
+    })
 
-      describe('and a second decimal point is typed after the first', () => {
+    describe('and the user types only a minus sign', () => {
+      beforeEach(() => {
+        userEvent.type(input, '-')
+      })
+
+      assertInputValue('-')
+      assertOnChangeCall(NaN, 1)
+
+      describe('and the increase button is clicked', () => {
         beforeEach(() => {
           onChangeSpy.mockReset()
-          userEvent.type(input, '.25')
+          wrapper.getByTestId('number-input-increase').click()
         })
 
-        assertInputValue('100.125')
-        assertOnChangeCall(100.125, 2)
+        assertInputValue('-')
+
+        it('does not call the onChange callback', () => {
+          expect(onChangeSpy).not.toHaveBeenCalled()
+        })
       })
 
-      describe('and a second decimal point is typed before the first', () => {
```

Comment:
This was removed and replaced with an equivalent Cypress test as user-event's `type`/`keyboard` functions aren't realistic enough to trigger `onBeforeInput`.